deploy.tf

```diff
@@ -306,70 +128,49 @@ resource "digitalocean_droplet" "k8s_worker" {
     region = "${var.do_region}"
     size = "${var.size_worker}"
     private_networking = true
-    user_data = "${data.template_file.worker_yaml.rendered}"
+    # user_data = "${data.template_file.worker_yaml.rendered}"
     ssh_keys = ["${split(",", var.ssh_fingerprint)}"]
+    depends_on = ["digitalocean_droplet.k8s_master"]
 
-
-
-    # Generate k8s_worker client certificate
-    provisioner "local-exec" {
-        command = <<EOF
-            ${path.module}/cfssl/generate_client.sh k8s_worker
-EOF
-    }
-
-    # Provision k8s_master client certificate
-    provisioner "file" {
-        source = "./secrets/ca.pem"
-        destination = "/home/core/ca.pem"
-        connection {
-            type = "ssh",
-            user = "core",
-            private_key = "${file(var.ssh_private_key)}"
-        }
-    }
+    # Start kubelet
     provisioner "file" {
-        source = "./secrets/client-k8s_worker.pem"
-        destination = "/home/core/worker.pem"
+        source = "./02-worker.sh"
+        destination = "/tmp/02-worker.sh"
         connection {
             type = "ssh",
             user = "core",
             private_key = "${file(var.ssh_private_key)}"
         }
     }
+
     provisioner "file" {
-        source = "./secrets/client-k8s_worker-key.pem"
-        destination = "/home/core/worker-key.pem"
+        source = "./install-kubeadm.sh"
+        destination = "/tmp/install-kubeadm.sh"
         connection {
             type = "ssh",
             user = "core",
             private_key = "${file(var.ssh_private_key)}"
         }
     }
 
-    # TODO: permissions on these keys
-    provisioner "remote-exec" {
-        inline = [
-            "sudo mkdir -p /etc/kubernetes/ssl",
-            "sudo cp /home/core/{ca,worker,worker-key}.pem /etc/kubernetes/ssl/.",
-            "sudo mkdir -p /etc/ssl/etcd/",
-            "sudo mv /home/core/{ca,worker,worker-key}.pem /etc/ssl/etcd/."
-        ]
+    provisioner "file" {
+        source = "./secrets/kubeadm_join"
+        destination = "/tmp/kubeadm_join"
         connection {
             type = "ssh",
             user = "core",
             private_key = "${file(var.ssh_private_key)}"
         }
     }
 
-    # Start kubelet
+    # Install dependencies and join cluster
     provisioner "remote-exec" {
         inline = [
-            "sudo systemctl daemon-reload",
-            "sudo systemctl start flanneld",
-            "sudo systemctl enable flanneld",
-            "sudo systemctl start kubelet",
-            "sudo systemctl enable kubelet"
+            "chmod +x /tmp/install-kubeadm.sh",
+            "sudo /tmp/install-kubeadm.sh",
+            "export NODE_PRIVATE_IP=\"${digitalocean_droplet.k8s_worker.ipv4_address}\"",
```

Comment:
This syntax causes an error for me. for me the fix was to change line 171 to `"export NODE_PRIVATE_IP=\"${digitalocean_droplet.k8s_worker.0.ipv4_address}\"",`. Also `NODE_PRIVATE_IP` is misleading as `worker.0.ipv4_address` is the public IP. Preferably this should be `worker.0.ipv4_address_private`.